### PR TITLE
widen dependency tool support

### LINKED ISSUE #18: openwrt/gl.inet devices limited binary support
can fallback to `od` instead of `hexdump`
also curl might not support `--compressed`, still works ok

### PR DESCRIPTION
**Pull request by TheNicholasNick**
gl.inet device has limited binaries.

closes #18

### CODE REVIEW

File: warp.sh [hunk 1]
```diff
@@ -135,8 +159,8 @@ else
 	# Register a new account if no refresh token is provided
 	wg_private_key="$(wg genkey)"
 	wg_public_key="$(printf %s "${wg_private_key}" | wg pubkey)"
-	reg="$(cfcurl --header 'Content-Type: application/json' --request "POST" --header 'CF-Access-Jwt-Assertion: '"${teams_ephemeral_token}" \
-		--data '{"key":"'"${wg_public_key}"'","install_id":"","fcm_token":"","model":"","serial_number":"","locale":"en_US"}' \
+	reg="$(cfcurl --header 'Content-Type: application/json' --request "POST" --header "CF-Access-Jwt-Assertion: ${teams_ephemeral_token}" \
+		--data '{"key":"'"${wg_public_key}"'","install_id":"","fcm_token":"","model":"","serial_number":"","name":"","locale":"en_US"}' \
```
Comment (rany2) [inline warp.sh:163]: What is this for?

Comment (TheNicholasNick) [reply warp.sh:163]: it came in as part #17 and i figured this was a formatting fix that worked - just added the empty name...

think need to autogen the name somehow or provide the option as a param

**Comment by TheNicholasNick** [comment]
am failing to sign this commit as don't want to figure out gpg and ssh key is going to be deleted in a moment...

**Comment by rany2** [comment]
Closes as it this PR didn't work right with `od`. I had to strip the leading whitespace as well otherwise in some cases the clientid that is generated is incorrect or the script errors out. See: https://github.com/rany2/warp.sh/pull/21

**Pull request closed without merging**